feat(curriculum): add prime number generator Python (#51370)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-concurrency-by-building-a-prime-number-generator/64e750d9e426ffe0a7c40dfa.md

---
id: 64e750d9e426ffe0a7c40dfa
title: Step 1

> challengeType: 0

Reviewer: ```
challengeType: 20
```

dashedName: step-1
---

# --description--

Asynchronous programming is a programming paradigm that allows tasks to run concurrently, without blocking the execution of other tasks.

In this challenge you would learn how concurrency works by building a prime number generator.

> To start, import the `asyncio` library, which provides tools for working with asynchronous code.

Reviewer 2: ```suggestion
To start, import the built-in `asyncio` module, which provides tools for working with asynchronous code.
```
Python usually uses the _module_ instead of library.

# --hints--

> You should use the `import` keyword to import the `asyncio` library.

Reviewer 2: ```suggestion
You should use the `import` keyword to import the `asyncio` module.
```

```js
({ test: () =>  assert.equal(__userGlobals.get('import'), '{}') })
```

> You should use the `asyncio` keyword.

Reviewer 2: ```suggestion
You should import the `asyncio` module.
```
